src/zip_code_tree.hpp

```diff
@@ -691,4 +872,758 @@ struct iterator_traits<vg::ZipCodeTree::reverse_iterator>{
 
 }
 
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+/// Implementations for the templated functions using Minimizers since the definition is in the minimizer_mapper
+//TODO: This really shouldn't be in the hpp file
```

Comment:
Can this move to the CPP?

Reply:
Maybe it would be better to move the `Minimizer` class (and maybe `Seed`, `Cluster`, etc) to a separate file?

Reply:
I think you want to put the template declaration in the header, and the template definition in the CPP, but also explicitly instantiate it for the right minimizer type with a line in the CPP:

```
template void void ZipCodeForest::fill_in_forest<MinimizerMapper::Minimizer>(const vector<Seed>&, const VectorView<MinimizerMapper::Minimizer>&, const SnarlDistanceIndex&, size_t, size_t);
```

I think you can also leave the variable names in?

Then it knows to actually make code for that instantiation of the template in the `.o` file that comes from that CPP file, and then the linker is happy.

Reply:
Thanks! I could not figure out how to get it to work in the CPP file